src/mtl.rs

```diff
@@ -279,9 +350,70 @@ impl Mtl {
         }
 
         if let Some(material) = material {
-            mtl.materials.push(material);
+            self.materials.push(Arc::new(material));
         }
 
-        Ok(mtl)
+        Ok(self)
+    }
+
+    pub fn write_to_buf<W: Write>(&self, out: &mut W) -> Result<(), io::Error> {
```

Comment:
nit: these days it's a tiny bit easier to write as `(&self, out: &mut impl Write)`